do.sh

```diff
@@ -1,5 +1,6 @@
 #!/usr/bin/env bash
 
+CALLER_PWD=$PWD
```

Comment:
Just curious, why do it this way?

Reply:
Do you see a better way?  One day I'll murder `do.sh`

Reply:
Didn't understand the intent, just noticed your first commit description

Reply:
Let me break the `do.sh` change into a different PR, it doesn't belong in here :)